api/src/main/java/com/exelaration/abstractmemery/services/implementations/FileStorageServiceImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.exelaration.abstractmemery.services.implementations;
+
+import com.exelaration.abstractmemery.services.FileStorageService;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.springframework.stereotype.Service;
+
+@Service("fileStorageService")
+public class FileStorageServiceImpl implements FileStorageService {
+  private static final String uploadingDir = "/app/src/main/resources/images/";
+
+  public String save(String fileName, InputStream fileStream) {
+    String fileLocation = "";
+    createDirectory();
+    if (fileName.equals("")) {
+      throw new RuntimeException("You must select the a file for uploading");
+    }
+    try {
+      fileLocation = (uploadingDir + fileName);
```

Comment:
We don't need the full file path of the image location, just the name.  We'll use the name to have the file storage service find the image for us later.

Reply:
Right now we are saving the full path under `file_location` and just the name under `file_name`. If we start only saving the name under `file_location`, should what we have as `file_name` in the db be changed to something like `image_title` and put unique constraints on the actual name of the file? Right now, if someone uploaded two pictures both called `cat.jpg`, the second would just save over the first. Could we potentially change the name of the second to `cat-2.jpg` and have its title still be `Cat`?